### PR TITLE
Update to 1.21.4 + fog adjustment

### PR DESCRIPTION
**Pull request by Libreh**
I updated the mod to 1.21.4 and made disabling the terrain fog not hide clouds (unlike before), let me know what you think!

### CODE REVIEW

File: build.gradle.kts
Comment (Bactrosaurus) [inline build.gradle.kts:1]: * Update Kotlin plugins jvm and serialization to version 2.1.0
* Update fabric-language-kotlin to 1.13.0+kotlin.2.1.0
* Update Mod-Menu to 13.0.0-beta.1 (adds Support for 1.21.4)

File: gradle/wrapper/gradle-wrapper.properties
Comment (Bactrosaurus) [inline gradle/wrapper/gradle-wrapper.properties:1]: * Update to Gradle 8.11.1 with gradle wrapper --gradle-version 8.11.1

File: src/main/java/de/daniel/bactromod/mixins/features/boatmap/MixinItemInHandRenderer.java [hunk 1]
```diff
@@ -48,7 +48,7 @@ private float clampMainHandHeight(float f, float g, float h) {
         ItemStack itemStack = getLocalPlayer().getMainHandItem();
         boolean mapInMainHand = this.mainHandItem.is(Items.FILLED_MAP);
         if (isOptionDisabled() || !mapInMainHand) return Mth.clamp(f, g, h);
-        float i = getLocalPlayer().getAttackStrengthScale(1.0F);
+        float i = getLocalPlayer().getAttackAnim(1.0F);
```
Comment (Bactrosaurus) [inline src/main/java/de/daniel/bactromod/mixins/features/boatmap/MixinItemInHandRenderer.java]: Why? Look at target method call. We want to replicate behaviour of else branch which uses getAttackStrengthScale.

File: src/main/java/de/daniel/bactromod/mixins/features/windowborder/MixinWindow.java [hunk 2]
```diff
@@ -25,7 +24,7 @@ public class MixinWindow {
     private long window;
 
     @Inject(method = "<init>", at = @At(value = "TAIL"))
-    private void init(WindowEventHandler handler, ScreenManager manager, DisplayData display, String videoMode, String title, CallbackInfo ci) {
+    private void init(WindowEventHandler windowEventHandler, ScreenManager screenManager, DisplayData displayData, String string, String string2, CallbackInfo ci) {
```
Comment (Bactrosaurus) [inline src/main/java/de/daniel/bactromod/mixins/features/windowborder/MixinWindow.java]: Don't use generic names like string or string2, even if values are not used.

File: src/main/resources/fabric.mod.json
Comment (Bactrosaurus) [inline src/main/resources/fabric.mod.json:1]: * Update fabric-language-kotlin to 1.13.0+kotlin.2.1.0

**Review by Bactrosaurus — CHANGES_REQUESTED** [review]
Changes with no comments are ok.

**Comment by Libreh** [comment]
Thanks for the feedback!
How about now?

**Comment by Bactrosaurus** [comment]
Thank you for fixing the minor issues! New version will be released on Modrinth very soon!

**Pull request merged**